src/Grpc.AspNetCore.Server/InterceptorRegistration.cs

```diff
@@ -0,0 +1,49 @@
+﻿#region Copyright notice and license
+
+// Copyright 2018 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+
+namespace Grpc.AspNetCore.Server
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class InterceptorRegistration
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="args"></param>
+        internal InterceptorRegistration(Type type, object[] args)
+        {
+            Type = type;
+            Args = args;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public Type Type { get; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        internal object[] Args { get; }
```

Comment:
Why internal? You could internally save as `object[]` and publicly expose as `IReadOnlyList<object>` to keep it immutable.

Reply:
I was thinking about this and couldn't come up with a use where the arguments would need to be inspected so I took the approach of not exposing unless needed.